lib/hexo/post.ts

```diff
@@ -84,13 +84,14 @@ class PostRenderEscape {
 
       if (state === STATE_PLAINTEXT) { // From plain text to swig
         if (char === '{') {
-          if (next_char === '{') {
+          // check if it is a complete tag {{ }}
+          if (next_char === '{' && /\{\{.+?\}\}/.test(str)) {
             state = STATE_SWIG_VAR;
             idx++;
-          } else if (next_char === '#') {
+          } else if (next_char === '#' && /\{#.+?#\}/.test(str)) {
```

Comment:
With the change, we will run `regexp.test` again and again for every character inside `str`.

The `.test` should be hoist to the top of the function.

Reply:
thanks, I have updated it